ssc/common_financial.cpp

```diff
@@ -1221,29 +1204,43 @@ bool dispatch_calculations::setup()
 		throw general_error(m_error);
 	}
 
-	m_periods.resize(8760, 1);
+    m_dispatch_tod_factors = m_cm->as_vector_double("dispatch_tod_factors");
+    if (m_dispatch_tod_factors.size() != 9)
+        return false;
+
+    m_periods.resize(8760, 1);
 	ssc_number_t *ppa_multipliers = m_cm->allocate("ppa_multipliers", 8760);
 
 	for (int i = 0; i < 8760; i++)
 	{
 		m_periods[i] = tod[i];
 
+        if ((tod[i] >= 1) && (tod[i] <= 9)) {
+
```

Comment:
Does this generate a warning? If so, please transition to a "not" format. If not, please add `// Nothing to do` so it's clear this was intentional.

Reply:
Cleaned up and removed else statement.